src/plot_api/subroutines.js

```diff
@@ -54,8 +54,8 @@ function lsInner(gd) {
     var i, subplot, plotinfo, xa, ya;
 
     fullLayout._paperdiv.style({
-        width: (fullLayout.autosize) ? '100%' : fullLayout.width + 'px',
-        height: (fullLayout.autosize && !fullLayout._hasZeroHeight) ? '100%' : fullLayout.height + 'px'
+        width: (fullLayout.autosize && !gd._context._hasZeroWidth && !gd.layout.width) ? '100%' : fullLayout.width + 'px',
+        height: (fullLayout.autosize && !gd._context._hasZeroHeight && !gd.layout.height) ? '100%' : fullLayout.height + 'px'
```

Comment:
The `&& !gd.layout.width` and `&& !gd.layout.height` conditions are there to ensure that explicit width/height wins over `autosize:true`.

Reply:
I just realized that one way of making sure we're not breaking anything out there would be to add a `&& gd._context.responsive`. That way, we would absolutely have no regression for people not using `responsive`! After all, setting the container's size to 100% is only useful for responsiveness in modern CSS layouts.

Reply:
> I just realized that one way of making sure we're not breaking anything out there would be to add a `&& gd._context.responsive`

:+1: